modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java

```diff
@@ -3962,11 +3963,19 @@ public ImageWorker affine(AffineTransform tx, Interpolation interpolation, doubl
                                 newROI = (ROI) property;
                             }
                             setROI(newROI);
-                            paramBlk.add(warp).add(interpolation).add(bgValues).add(newROI);
+                            paramBlk.add(warp).add(interpolation).add(newROI);
                             if (oldNoData != null) {
                                 paramBlk.set(oldNoData, 4);
                             }
                         }
+                        
+                        // handle background values
+                        if(bgValues == null && sourceParamBlock.getNumParameters() > 2) {
```

Comment:
"if(condition)" should be "if (condition)" with a space between "if" and the "("
(Formatting rules)